datafusion/core/tests/physical_optimizer/enforce_distribution.rs

```diff
@@ -1783,6 +1783,7 @@ fn union_to_interleave() -> Result<()> {
     );
 
     //  Union
+    #[allow(deprecated)]
```

Comment:
Let's have an issue to clean these up and add  a `// TODO (issue link) resolve deprecation` comment

Reply:
- Filed https://github.com/apache/datafusion/issues/17559